apps/browser-extension-wallet/src/views/browser-view/components/Layout/Layout.tsx

```diff
@@ -56,11 +57,19 @@ export const Layout = ({ children, drawerUIDefaultContent, isFullWidth }: Layout
   }, [backgroundServices, setTheme]);
 
   useEffect(() => {
-    const timer = setTimeout(() => {
-      setShowPinExtension(false);
-    }, PIN_EXTENSION_TIMEOUT);
-    return () => clearTimeout(timer);
-  }, [setShowPinExtension]);
+    let timer: NodeJS.Timeout;
+    if (!showDappBetaModal) {
+      timer = setTimeout(() => {
+        setShowPinExtension(false);
+      }, PIN_EXTENSION_TIMEOUT);
+    }
+
+    return () => {
+      if (timer) {
+        clearTimeout(timer);
+      }
+    };
```

Comment:
We should not use `NodeJS.Timeout` type, as it is not a nodejs runtime. Please use `number` and `window.setTimeout` or consider this change:
```suggestion
    if (showDappBetaModal) return undefined;
    const timer = window.setTimeout(() => {
      setShowPinExtension(false);
    }, PIN_EXTENSION_TIMEOUT);
    return () => window.clearTimeout(timer);
```

Reply:
Fixed on a80bda7. Took this approach to avoid the "eslint: consistent-return" 😅

Reply:
But then, we would have to fix "ESLint: Do not use useless `undefined`.(unicorn/no-useless-undefined)", and `eslint --fix` automatically removes the undefined. 😄

Reply:
I think disabling `eslint: consistent-return` would work perfectly in this case.

Reply:
I don't like those `unicorn` rules. I noticed, they complicating the development...

Reply:
I do really like unicorn rules. They generally make the code cleaner. :)